src/Sarif/Writers/SarifLogger.cs

```diff
@@ -15,48 +15,52 @@
 
 namespace Microsoft.CodeAnalysis.Sarif.Writers
 {
-    public class SarifLogger : IDisposable, IAnalysisLogger
+    public class SarifLogger : BaseLogger, IDisposable, IAnalysisLogger
     {
         private readonly Run _run;
         private readonly TextWriter _textWriter;
         private readonly bool _persistArtifacts;
         private readonly bool _closeWriterOnDispose;
-        private readonly LoggingOptions _loggingOptions;
+        private readonly LogFilePersistenceOptions _logFilePersistenceOptions;
         private readonly JsonTextWriter _jsonTextWriter;
         private readonly OptionallyEmittedData _dataToInsert;
         private readonly OptionallyEmittedData _dataToRemove;
         private readonly ResultLogJsonWriter _issueLogJsonWriter;
         private readonly InsertOptionalDataVisitor _insertOptionalDataVisitor;
 
-        protected const LoggingOptions DefaultLoggingOptions = LoggingOptions.PrettyPrint;
+        protected const LogFilePersistenceOptions DefaultLogFilePersistenceOptions = LogFilePersistenceOptions.PrettyPrint;
 
         public SarifLogger(
             string outputFilePath,
-            LoggingOptions loggingOptions = DefaultLoggingOptions,
+            LogFilePersistenceOptions logFilePersistenceOptions = DefaultLogFilePersistenceOptions,
             OptionallyEmittedData dataToInsert = OptionallyEmittedData.None,
             OptionallyEmittedData dataToRemove = OptionallyEmittedData.None,
             Tool tool = null,
             Run run = null,
             IEnumerable<string> analysisTargets = null,
             IEnumerable<string> invocationTokensToRedact = null,
             IEnumerable<string> invocationPropertiesToLog = null,
-            string defaultFileEncoding = null)
+            string defaultFileEncoding = null,
+            IEnumerable<FailureLevel> level = null,
```

Comment:
>level [](start = 38, length = 5)

nit: rename to levels and kinds

Reply:
Done.

---
In reply to: [561430439](https://github.com/microsoft/sarif-sdk/pull/2241#discussion_r561430439) [](ancestors = 561430439)